rpc-client/src/rpc_client.rs

```diff
@@ -1900,6 +1900,21 @@ impl RpcClient {
         self.invoke((self.rpc_client.as_ref()).wait_for_max_stake(commitment, max_stake_percent))
     }
 
+    pub fn wait_for_max_stake_below_threshold_with_timeout(
+        &self,
+        commitment: CommitmentConfig,
+        max_stake_percent: f32,
+        timeout: Option<Duration>,
```

Comment:
If someone is specifically using the `_with_timeout()` variant when there is a no-timeout variant as well, I would think they'd have a timeout in mind and the parameter should be `Duration`, not `Option<Duration>`.

As an example, another function from this file:
https://github.com/anza-xyz/agave/blob/ec781d3b5aa17dbcc889bf4c8c9d2513dde15cd9/rpc-client/src/rpc_client.rs#L297-L302